jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -587,10 +586,7 @@ ecma_builtin_array_prototype_object_push (ecma_value_t this_arg, /**< this argum
                     ecma_builtin_array_prototype_helper_set_length (obj_p, n),
                     ret_value);
 
-    ecma_number_t *ret_num_p = ecma_alloc_number ();
-    *ret_num_p = n;
-
-    ret_value = ecma_make_number_value (ret_num_p);
+    ret_value = ecma_make_number_value (n);
```

Comment:
Something seems to be sub-optimal here. Some lines above, we have
`ecma_number_t n = ((ecma_number_t) ecma_number_to_uint32 (length_var));`
Is it intentional to convert an uint32 to float first and then most probably convert it back from float?
Or, is it because of this line
`ecma_string_t *n_str_p = ecma_new_ecma_string_from_number (n);`?
I'm not sure right now whether we have a quick(er) implementation for uint32-to-string conversion.